### PR TITLE
feat(project)添加前端项目

### PR DESCRIPTION
**Pull request by haungrui777**
本代码是iotdb-web-workbench前端0.12.1代码，主要功能如下图：
avatar
请审核。

### CODE REVIEW

File: frontend/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# iotdb-fe
```
Comment (qiaojialin) [inline frontend/README.md:1]: need license

File: frontend/src/App.vue [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/* * Licensed to the Apache Software Foundation (ASF) under one * or more contributor license agreements. See the NOTICE file * distributed with this work for additional information * regarding
```
Comment (qiaojialin) [inline frontend/src/App.vue:1]: better to format this

File: frontend/README.md (hunk 1 above)
Comment (haungrui777) [reply frontend/README.md:20]: done

File: frontend/src/App.vue (hunk 2 above)
Comment (haungrui777) [reply frontend/src/App.vue]: done

**Comment by NiklasMerz** [comment]
A good way to check code for Apache headers is to use [Apache RAT](https://creadur.apache.org/rat/). 

For this PR I get the following output:

```
java -jar ~/apache-rat-0.13.jar .

*****************************************************
Summary
-------
Generated at: 2021-09-11T10:30:33+02:00

Notes: 2
Binaries: 6
Archives: 0
Standards: 73

Apache Licensed: 62
Generated Documents: 0

JavaDocs are generated, thus a license header is optional.
Generated files do not require license headers.

11 Unknown Licenses

*****************************************************

Files with unapproved licenses:

  ./.gitignore
  ./NOTICE-binary
  ./frontend/.browserslistrc
  ./frontend/.eslintignore
  ./frontend/.gitignore
  ./frontend/package-lock.json
  ./frontend/package.json
  ./frontend/public/iconfont.js
  ./frontend/src/shims-vue.d.ts
  ./frontend/src/store/index.js
  ./frontend/tests/unit/example.spec.js

*****************************************************

*****************************************************
```

Some of those files are probably fine and they can be ignored (RAT has a .ratignore file for this).  But I would check the `.js` files again.

I also had a brief look at the dependency licenses yarn gives me with `yarn licenses list` gives me and compared it with [Apaches policy](https://www.apache.org/legal/resolved.html). I did not find any issues so far.

**Comment by JulianFeinauer** [comment]
I added PR #4 which is this branch + several improvements and fixes (and some rebases after mergint #1 / #3). So I will close this one, I hope nobody minds?

**Pull request merged**